Tensile/KernelWriterAssembly.py

```diff
@@ -4901,24 +5072,144 @@ def lraFinalOffset(self, kernel, tP):
     rReg = self.vgprPool.checkOut(1) # remainder, unused here
     dividendReg = "Serial"
     tmpVgpr = self.vgprPool.checkOut(2,"tmpVgpr")
+    tmpVgpraddr = self.vgprPool.checkOut(1,"tmpVgpraddr")
+    tmpVgprPadoffset = self.vgprPool.checkOut(1,"tmpVgprPadoffset")
     tmpSgpr = self.getTmpSgpr(1)
-    kStr += vectorStaticDivideAndRemainder(qReg, rReg, dividendReg, divisor, \
-        tmpVgpr, tmpSgpr)
+    if not kernel["TransposeLDS"] == 1:
+      kStr += vectorStaticDivideAndRemainder(qReg, rReg, dividendReg, divisor, \
+          tmpVgpr, tmpSgpr)
     sgid = qReg
 
     tIdx = tP["tensorIdx"]
     if kernel["MatrixInstruction"]:
-      if tc == "A": # For BBlocks, A and B use this case
-        kStr += inst("s_mov_b32", \
-            sgpr(tmpSgpr), \
-            hex(kernel["MacroTile%u"%tIdx] + kernel["LdsPad%s"%tc]), \
-            "MT%u+PAD"%tIdx )
-      else: # For BBlocks, don't use else case
-        kStr += inst("s_mov_b32", \
-            sgpr(tmpSgpr), \
-            #hex(kernel["MacroTile%u"%tIdx] + kernel["LdsPad%s"%tc]), \
-            hex((kernel["MacroTile%u"%tIdx] + kernel["LdsPad%s"%tc]) // 4), \
-            "MT%u+PAD"%tIdx )
+      if kernel["TransposeLDS"]:
+        #TransposeLDS feature supports LDS memory format is same as global memory format when TLU=0
+        if tc == "A": # For BBlocks, A and B use this case
+          #re-calculate tileA assignment  remove this once tileAssignmentB bug is fixed
+          kStr += inst("v_and_b32", \
+              vgpr(tP["gpr"]["lro"]), \
+              hex(kernel["SubGroup0"]-1), \
+              vgpr("Serial"), \
+              "vectorStaticDiv: %s = %s %% %u" \
+              % (vgpr(tP["gpr"]["lro"]), vgpr("Serial"), (kernel["SubGroup0"]-1)))
+        else:
+          kStr += inst("v_and_b32", \
+              vgpr(tP["gpr"]["lro"]), \
+              hex((((kernel["MatrixInstB"]*kernel["MatrixInstM"])//kernel["SubGroup0"])*kernel["MatrixInstN"])-1), \
```

Comment:
(M*N*B)//MIWG0
Maybe make a variable for this calculation? It is similar to one done in a number of other places. Grouping MNB is easier to read. Also, I think we've used MIWG for MI calculations?